src/components/Admin/User/UserPermissionModal.tsx

```diff
@@ -0,0 +1,132 @@
+                  : p.permission?.permission === 'write',
+              index,
+            };
+          }) ?? [],
+      }));
+    },
+  });
+
+  const handleSubmit = (values: FormType) => {};
+
+  return (
+    <form onSubmit={onSubmit(handleSubmit)}>
+      <Table>
+        <thead>
+          <tr>
+            <th>Dashboard</th>
+            <th>
+              <Text align="center">Can visit</Text>
+            </th>
+            <th>
+              <Text align="center">Can modify</Text>
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {values.permissions?.map((p, index) => (
+            <tr>
+              <td>
+                <Text weight={500}>{p.id}</Text>
+              </td>
+              <td>
+                <Center>
+                  <Checkbox
+                    {...getInputProps(`permissions.${index}.canRead`, { type: 'checkbox' })}
+                    onChange={(ev) => {
+                      getInputProps(`permissions.${index}.canRead`, { type: 'checkbox' }).onChange(
+                        ev
+                      );
+                      if (!ev.target.checked) {
+                        setFieldValue(`permissions.${index}.canWrite`, false);
+                      }
+                    }}
+                  />
+                </Center>
+              </td>
+              <td>
+                <Center>
+                  <Checkbox
+                    {...getInputProps(`permissions.${index}.canWrite`, { type: 'checkbox' })}
+                    onChange={(ev) => {
+                      getInputProps(`permissions.${index}.canWrite`, { type: 'checkbox' }).onChange(
+                        ev
+                      );
+                      if (ev.target.checked) {
+                        setFieldValue(`permissions.${index}.canRead`, true);
+                      }
+                    }}
+                  />
+                </Center>
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+      <Group position="right">
+        <Button onClick={() => context.closeModal(id)}>Cancel</Button>
+        <Button type="submit">Save</Button>
+      </Group>
+    </form>
+  );
+};
+
+export const openUserPermissionModal = (innerProps: InnerProps) =>
+  openContextModal({
+    modal: 'userPermissionModal',
+    title: <Title order={4}>Modify the permissions of {innerProps.user.username}</Title>,
+    innerProps,
+  });
+
+type UseUserPermissionsQueryResult = {
+  id: string;
+  permission: (DashboardPermission & { permission: 'read' | 'write' }) | null;
+}[];
+
+const useUserPermissionsQuery = (
+  id: string,
+  { onSuccess }: { onSuccess: ((data: UseUserPermissionsQueryResult) => void) | undefined }
+) =>
+  useQuery<UseUserPermissionsQueryResult>({
+    queryKey: ['user/permission', { id }],
+    queryFn: async () => {
+      const result = await axios.get(`/api/users/${id}/permissions`);
+      return result.data;
+    },
+    onSuccess,
```

Comment:
Don't forget to adjust cache time and stale time correctly from the base config when needed